src/town_gui.cpp

```diff
@@ -698,7 +727,6 @@ struct TownDirectoryWindow : public Window {
 
 			this->towns.shrink_to_fit();
 			this->towns.RebuildDone();
-			this->vscroll->SetCount((uint)this->towns.size()); // Update scrollbar as well.
```

Comment:
What's this change doing here?

Reply:
Removed that line, because we no longer use the vscroll in this window, so the line does nothing.

Reply:
I don't follow. This window is the town directory window, which definitely still uses the scrollbar

Reply:
To be clear, _this_ is the scrollbar you made non-functional by deleting that line:
![image](https://user-images.githubusercontent.com/1062071/187780064-145324fc-86e5-472d-9cc5-38d8936abca6.png)

And to be extra clear, this is the same generated world, just with the window made larger to fit more towns. It still has many more, but you can't view the list because the scroll bar doesn't work.
![image](https://user-images.githubusercontent.com/1062071/187780178-031d8e18-69bc-42a7-bb27-cc32075fd2a4.png)

Reply:
Yep very true, got lost in the sauce and didn't realise what object I was working in. Fixed.